reference/api/settings.md

```diff
@@ -199,3 +402,992 @@ All settings will be reset to their default value.
+:::
+
+### Get stop words
+
+<RouteHighlighter method="GET" route="/indexes/{index_uid}/settings/stop-words" />
+
+Get the stop words list of an index.
+
+#### Path parameters
+
+| Name        | Type   | Description                                                               |
+| :---------- | :----- | :------------------------------------------------------------------------ |
+| **`uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |
+
+#### Example
+
+<CodeSamples id="get_stop_words_1" />
+
+##### Response: `200 Ok`
+
+```json
+[
+  "of",
+  "the",
+  "to"
+]
+```
+
+### Update stop words
+
+<RouteHighlighter method="PUT" route="/indexes/{index_uid}/settings/stop-words" />
+
+Update the list of stop words of an index. This will re-index all documents in the index.
+
+#### Path parameters
+
+| Name        | Type   | Description                                                               |
+| :---------- | :----- | :------------------------------------------------------------------------ |
+| **`uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |
+
+#### Body
+
+An array of strings that contains the stop words.
+
+If a list of stop words already exists, it will be overwritten (_replaced_).
+
+#### Example
+
+<CodeSamples id="update_stop_words_1" />
+
+##### Response: `202 Accepted`
+
+```json
+{
+    "taskUid": 1,
+    "indexUid": "movies",
+    "status": "enqueued",
+    "type": "settingsUpdate",
+    "enqueuedAt": "2021-08-11T09:25:53.000000Z"
+}
+```
+
+You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).
+
+### Reset stop words
+
+<RouteHighlighter method="DELETE" route="/indexes/{index_uid}/settings/stop-words" />
+
+Reset the list of stop words of an index to its default value.
+
+#### Path parameters
+
+| Name        | Type   | Description                                                               |
+| :---------- | :----- | :------------------------------------------------------------------------ |
+| **`uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |
+
+#### Example
+
+<CodeSamples id="reset_stop_words_1" />
+
+##### Response: `202 Accepted`
+
+```json
+{
+    "taskUid": 1,
+    "indexUid": "movies",
+    "status": "enqueued",
+    "type": "settingsUpdate",
+    "enqueuedAt": "2021-08-11T09:25:53.000000Z"
+}
+```
+
+You can use this `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).
+
+## Synonyms
+
+The `synonyms` object contains words and their respective synonyms. A synonym in Meilisearch is considered equal to its associated word in a search query. This route allows you to manage synonyms for an index.
```

Comment:
I would like to shorten this paragraph to 1–2 brief, efficient sentences. This is just a suggestion, so feel free to rewrite it!

```suggestion
The `synonyms` object contains words and their respective synonyms. A synonym in Meilisearch is considered equal to its associated word for the purposes of calculating search results.
```